js/index.js

```diff
@@ -137,6 +137,94 @@ Metrics.prototype = {
+                                          event_label,    // Metric label (e.g., 'memory')
+                                          event_value) {  // Value of float (float)
+        var self = this;
+        var event_string = formatEventString();
+        this.log("METRICS - event string:" + event_string);
+        var post_options = {
+            host: 'www.google-analytics.com',
+            port: '443',
+            path: '/batch',
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/x-www-form-urlencoded',
+                'Content-Length': event_string.length
+            }
+        };
+
+        var post_req = https.request(post_options, function(res) {
+            self.log("METRICS - request returned: " + res.statusCode);
+            self.log('METRICS - event recorded: ' + event_category + ', '
+                                                  + event_action +   ', '
+                                                  + event_label +    ', '
+                                                  + event_value);
+        });
+
+        // post the data
+        this.log("METRICS - Sending request...");
+        post_req.write(event_string);
+        post_req.end();
+
+        function formatEventString() {
+            encodeURIComponent(event_category);
+            encodeURIComponent(event_action);
+            encodeURIComponent(event_label);
+            encodeURIComponent(event_value);
+
+            encodeURIComponent(self.locale);
+            encodeURIComponent(self.os);
+            encodeURIComponent(self.os_version);
+            encodeURIComponent(self.device);
+            encodeURIComponent(self.app_name);
+            encodeURIComponent(self.app_version);
+            encodeURIComponent(self.app_update_channel);
+            encodeURIComponent(self.app_build_id);
+            encodeURIComponent(self.app_platform);
+            encodeURIComponent(self.arch);
+
+            var event_string = ('v=1&t=event&tid=UA-77033033-1&cid=' + self.clientId +
+                                '&ec=' + event_category +
+                                '&ea=' + event_action +
+                                '&el=' + event_label +
+                                '&ev=' + 1 +
+                                '&an=' + self.app_name +
+                                '&av=' + self.app_version +
+                                '&ul=' + self.locale +
+                                '&cd1=' + self.os +
+                                '&cd2=' + self.os_version +
+                                '&cd3=' + self.device +
+                                '&cd4=' + self.arch +
+                                '&cd5=' + self.app_platform +
+                                '&cd6=' + self.app_build_id +
+                                '&cd7=' + getFormattedTime()) +
+                                '&cd8=' + parseFloat(Math.round(event_value * 100) / 100).toFixed(2);
```

Comment:
Did you try 'parseFloat(event_value)'? Is the multiplication and division necessary?